src/main/java/com/cairnfg/waypoint/authorization/endpoints/account/RemoveCoClientEndPoint.java

```diff
@@ -37,98 +31,65 @@ public class RemoveCoClientEndPoint {
-   *
-   * @param accountId         The ID of the account to update.
-   * @param accountDetailsDto The DTO containing the details to update the account.
-   * @param principal         The principal object representing the currently authenticated user.
-   * @return A ResponseEntity containing the result of the account update operation.
-   */
   @Transactional
   @PatchMapping(PATH)
   @PreAuthorize("hasAnyAuthority('SCOPE_account.full', 'SCOPE_admin.full')")
   @Operation(
-      summary = "Removes Coclient from household. ",
-      description =
-          "Allows a user to remove a coclient from a household. Requires the `account.create` permission.",
+      summary = "Removes Co-client from household.",
+      description = "Removes a co-client account from the household and updates the primary contact if needed.",
       security = @SecurityRequirement(name = "oAuth2JwtBearer"),
       responses = {
-          @ApiResponse(responseCode = "200",
-              description = "OK - Account update was successful"),
+          @ApiResponse(responseCode = "200", description = "OK - Co-client removed successfully."),
           @ApiResponse(responseCode = "400", description = "Bad Request",
               content = {@Content(mediaType = "application/json",
                   schema = @Schema(implementation = ErrorMessage.class))}),
-          @ApiResponse(responseCode = "401", description = "Unauthorized",
-              content = {@Content(schema = @Schema(hidden = true))}),
-          @ApiResponse(responseCode = "403", description = "Forbidden",
-              content = {@Content(mediaType = "application/json",
-                  schema = @Schema(implementation = ErrorMessage.class))}),
           @ApiResponse(responseCode = "404", description = "Not Found",
               content = {@Content(mediaType = "application/json",
-                  schema = @Schema(implementation = ErrorMessage.class))}),
-          @ApiResponse(responseCode = "409", description = "Not Found",
-              content = {@Content(mediaType = "application/json",
-                  schema = @Schema(implementation = ErrorMessage.class))})})
+                  schema = @Schema(implementation = ErrorMessage.class))})
+      })
+  public ResponseEntity<?> removeCoClient(@PathVariable Long accountId, Principal principal) {
 
-  public ResponseEntity<?> updateHouseholdId(@PathVariable Long accountId,
-      @RequestBody UpdateAccountDetailsDto accountDetailsDto,
-      Principal principal) {
+    log.info("User [{}] is attempting to remove co-client for account [{}]", principal.getName(), accountId);
 
-    log.info("User [{}] is attempting to update householdId for account [{}]", principal.getName(),
-        accountId);
+    Account accountToRemove = accountService.getAccountById(accountId)
+        .orElseThrow(() -> new IllegalArgumentException("Account with ID [" + accountId + "] not found"));
```

Comment:
So the `orElseThrow()` approach you're adopting here is risky. It exposes exception to the caller. We should use the `generateFailureResponse()` approach laid out. It will allow us to control the error messages presented back to the caller

Reply:
It's fine to use `orElseThrow()` in the lower tiers of the application (specifically talking about the service layer) where you can catch the exception in the upper tier -- the controller layer. This way you can then handle the exception and present a controlled and intuitive response back to the caller.